src/hooks/useCurrentFaultTree.tsx

```diff
@@ -27,9 +28,14 @@ export const CurrentFaultTreeProvider = ({ faultTreeIri, children }: Props) => {
 
   const fetchFaultTree = async () => {
     try {
-      const { result, reqProb } = await faultTreeService.find(faultTreeIri);
+      const result = await faultTreeService.find(faultTreeIri);
       _setFaultTree(result);
-      if (reqProb) setRootReqProb(reqProb);
+      if (result.manifestingEvent.supertypes) {        
+        const failureRate = result.manifestingEvent?.supertypes[`${_NS_FTA_FMEA.PREFIX + "has-failure-rate"}`]
+        const requirement = failureRate[_NS_FTA_FMEA.PREFIX + "has-requirement"]
```

Comment:
Why not preparing your structures to be able to do:
`failureRate.requirement` ?

Reply:
updated